ocs_ci/deployment/disconnected.py

```diff
@@ -301,34 +380,77 @@ def mirror_index_image_via_oc_mirror(index_image, packages, icsp=None):
 
     if icsp:
         # update mapping.txt file with urls updated based on provided
-        # imageContentSourcePolicy
+        # imageContentSourcePolicy/imageDigestMirrorSet
         mapping_file = os.path.join(
             f"{mirroring_manifests_dir}",
             "mapping.txt",
         )
-        mirror_images_from_mapping_file(mapping_file, icsp)
 
-    # create ImageContentSourcePolicy
-    icsp_file = os.path.join(
+    if idms:
+        cmd += " --dry-run"
+        # running the command again with --dry-run to get mapping file
+        # so that we can pass it and get the failed mirrors
+        exec_cmd(cmd, timeout=18000)
+        mapping_file = os.path.join(
+            f"{mirroring_manifests_dir}",
+            "working-dir/dry-run/mapping.txt",
+        )
+        with open(mapping_file, "r") as file:
+            lines = file.readlines()
+
+        # Remove 'docker://' from each line
+        updated_lines = [line.replace("docker://", "") for line in lines]
+
+        with open(mapping_file, "w") as file:
+            file.writelines(updated_lines)
+
+    mirror_images_from_mapping_file(mapping_file, idms=idms)
```

Comment:
The `mirror_images_from_mapping_file` function is applicable only inside the `if icsp` and `if idms` conditions. If neither _icsp_ or _idms_ is defined (for example for Live/GA deployment), we don't need to use the workaround with mapping file and `oc image mirror ...` command as all the images should be properly mirrored via the standard `oc mirror` command.